src/autoIncrement.ts

```diff
@@ -95,17 +102,27 @@ export function AutoIncrementID(schema: mongoose.Schema<any>, options: AutoIncre
     if (!model) {
       logger.info('Creating idtracker model named "%s"', opt.trackerModelName);
       model = this.db.model(opt.trackerModelName, IDSchema, opt.trackerCollection);
+      model.findOne({ model: opt.model, field: opt.field } as AutoIncrementIDTrackerSpec).lean()
+        .then((counter: AutoIncrementIDTrackerSpec) => {
+          if (!counter) {
+            model.create({
+              model: opt.model,
```

Comment:
```suggestion
              model: (this.constructor as any).modelName,
```
this would be more dynamic and automatically gets the model name from the model the plugin was applied to, the only limitation i could find, was that no different connection than the parent could be used

Reply:
I agree with you. but the output of `getName(SomeClass)` is not as same as `mongoose collection` name. so we are getting duplicate entries in `identitycounters` collection.

Reply:
sorry i dont understand you, this plugin dosnt use the collection-name of the applied model, only the modelname

Reply:
actually we where following `mongoose's full lowercase and pluralize collection name` everywhere. so we are facing trouble. that's why we need this extra field. I fixed this by making the model field optional.

Reply:
could you give me a link to that, and explain why you want to store the collection name instead of the modelName?

PS: `getName(SomeClass)` is only meant when used with typegoose and its classes

Reply:
https://mongoosejs.com/docs/models.html#compiling

Reply:
our model name is `user` and same is given in `identitycounters model` field. but our class name starts with caps `class User` and so `getName(User)` returning `User` and duplicate entries got inserted as `model: 'User'`

Reply:
@nasr18, when you use typegoose's `getModelForClass` and dont use an custom name, then the model name is the same as the class, but the collection is still lowercased & pluralized

easy example:
```ts
// NodeJS: 12.15.0
// MongoDB: 4.2-bionic (Docker)
import { getModelForClass, prop } from "@typegoose/typegoose"; // @typegoose/typegoose@6.3.1
import * as mongoose from "mongoose"; // mongoose@5.8.11
import { getName } from "@typegoose/typegoose/lib/internal/utils";

class User {
  @prop()
  public username?: string;
}
const UserModel = getModelForClass(User);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verifyMASTER", useCreateIndex: true, useUnifiedTopology: true });

  console.log("User model name:", getName(User), UserModel.modelName); // log: User model name: User User
  console.log("User collection name:", UserModel.collection.name); // log: User collection name: users

  await mongoose.disconnect();
})();

```

and if you use some custom name (or automaticName turned on), then it will be the custom name and `getName` will return the correct one

but if you set the `collection` option, it dosnt affect the name, the collection name gets automaticly set based on the modelname (lowercased & pluralized) when no `collection` options is provided:
```ts
// NodeJS: 12.15.0
// MongoDB: 4.2-bionic (Docker)
import { getModelForClass, prop, modelOptions } from "@typegoose/typegoose"; // @typegoose/typegoose@6.3.1
import * as mongoose from "mongoose"; // mongoose@5.8.11
import { getName } from "@typegoose/typegoose/lib/internal/utils";

@modelOptions({ schemaOptions: { collection: "SomeCustomCollectionName" } })
class User {
  @prop()
  public username?: string;
}
const UserModel = getModelForClass(User);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verifyMASTER", useCreateIndex: true, useUnifiedTopology: true });

  console.log("User model name:", getName(User), UserModel.modelName); // log: User model name: User User
  console.log("User collection name:", UserModel.collection.name); // log: User collection name: SomeCustomCollectionName

  await mongoose.disconnect();
})();
```

that is why `this.constructor.modelName` is simpler and dosnt require user input

TL;DR: this plugin, `typegoose-auto-increment`'s function for `ID`'s is based on the *model name* and not on the *collection name*

Reply:
instead of passing `model` in `plugin`, passing `@modelOptions({ options: { customName: 'user' } })` fixed the problem.

Reply:
@nasr18, good to hear that you issue got fixed